src/runtime/semaphore_wrapper.cc

```diff
@@ -0,0 +1,47 @@
+/*!
+ *  Copyright (c) 2021 by Contributors
+ * \file semaphore_wrapper.cc
+ * \brief A simple corss platform semaphore wrapper
+ */
+#include "semaphore_wrapper.h"
```

Comment:
stl has semaphore. should we use it?

Reply:
stl's <semaphore> requires c++20, and my environment only supports c++11...